src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -0,0 +1,373 @@
+        #TODO Check if valid cluster
+        if cluster_id not in self.exports:
+            self.exports[cluster_id] = []
+
+        self.rados_namespace = cluster_id
+        if not self.check_fs(fs_name) or self.check_pseudo_path(pseudo_path):
+            return -errno.EINVAL,"", "Invalid CephFS name or export already exists"
+
+        user_id, key = self._create_user_key(cluster_id)
+        if isinstance(user_id, int):
+            return user_id, "", key
+        access_type = "RW"
+        if read_only:
+            access_type = "R"
+
+        ex_dict = {
+            'path': self.format_path(path),
+            'pseudo': self.format_path(pseudo_path),
+            'cluster_id': cluster_id,
+            'access_type': access_type,
+            'fsal': {"name": "CEPH", "user_id":cluster_id, "fs_name": fs_name, "sec_label_xattr": ""},
+            'clients': []
+            }
+
+        ex_id = self._gen_export_id()
+        export = Export.from_dict(ex_id, ex_dict)
+        export.fsal.cephx_key = key
+        self._save_export(export)
+
+        result = {
+            "bind": pseudo_path,
+            "fs": fs_name,
+            "path": path,
+            "cluster": cluster_id,
+            "mode": access_type,
+            }
+
+        return (0, json.dumps(result, indent=4), '')
+
+    def delete_export(self, ex_id):
+        raise NotImplementedError()
+
+    def make_rados_url(self, obj):
+        if self.rados_namespace:
+            return "rados://{}/{}/{}".format(self.rados_pool, self.rados_namespace, obj)
+        return "rados://{}/{}".format(self.rados_pool, obj)
+
+class NFSCluster:
+    def __init__(self, mgr, cluster_id):
+        self.cluster_id = "ganesha-%s" % cluster_id
+        self.pool_name = 'nfs-ganesha'
+        self.pool_ns = cluster_id
+        self.mgr = mgr
+
+    def create_empty_rados_obj(self):
+        common_conf = 'conf-nfs'
+        result = ''
+        with self.mgr.rados.open_ioctx(self.pool_name) as ioctx:
+            if self.pool_ns:
+                ioctx.set_namespace(self.pool_ns)
+            ioctx.write_full(common_conf, result.encode('utf-8'))
+            log.debug(
+                    "write configuration into rados object %s/%s/nfs-conf\n",
+                    self.pool_name, self.pool_ns)
+
+    def create_nfs_cluster(self, export_type, size):
+        if export_type != 'fs':
+            return -errno.EINVAL,"", "Invalid FSAL. Only CephFS is supported."
+
+        pool_list = [p['pool_name'] for p in self.mgr.get_osdmap().dump().get('pools', [])]
+        client = 'client.%s' % self.cluster_id
+
+        if self.pool_name not in pool_list:
+            r, out, err = create_pool(self.mgr, self.pool_name)
+            if r != 0:
+                return r, out, err
+            log.info("{}".format(out))
+
+            command = {'prefix': 'osd pool application enable', 'pool': self.pool_name, 'app': 'nfs'}
```

Comment:
No. Why do you want it in cephadm?

Reply:
In principle it might be good to let the backend manage the pool for recovery objects? I don't really have a strong feeling for where the line of responsibility should be drawn here though.

Reply:
I'm not entirely familiar with `osd pool application enable` .. what happens if this command is omitted? 

In either case, it does feel to like something that should be manged on the backend ...

Reply:
We get health warning on its omission. Is it fine if I remove this part before integrating cephadm in a new PR ?